modules/framework/galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/TestClassWrapper.java

```diff
@@ -228,33 +230,35 @@ public void runTestMethods(@NotNull ITestRunManagers managers, IDynamicStatusSto
             }     
         }
 
+        setResult(getResult(), managers);
+
         // Run @AfterClass methods
         for (GenericMethodWrapper afterClassMethod : afterClassMethods) {
             afterClassMethod.invoke(managers, this.testClassObject, null);
             if (afterClassMethod.fullStop()) {
-                if (this.result == null) {
-                    this.result = Result.failed("AfterClass method failed");
+                if (getResult() == null) {
+                    setResult( Result.failed("AfterClass method failed") , managers );
                 }
             }
         }
 
         try {
-            Result newResult = managers.endOfTestClass(this.result, null); // TODO pass the class level exception
+            Result newResult = managers.endOfTestClass(getResult(), null); // TODO pass the class level exception
             if (newResult != null) {
                 logger.info("Result of test run overridden to " + newResult.getName());
-                this.result = newResult;
+                setResult(newResult, managers);
             }
         } catch (FrameworkException e) {
             throw new TestRunException("Problem with end of test class", e);
         }
 
         // Test result
-        logger.info(LOG_ENDING + LOG_START_LINE + LOG_ASTERS + LOG_START_LINE + "*** " + this.result.getName()
+        logger.info(LOG_ENDING + LOG_START_LINE + LOG_ASTERS + LOG_START_LINE + "*** " + getResult().getName()
         + " - Test class " + testClass.getName() + LOG_START_LINE + LOG_ASTERS);
 
-        this.testStructure.setResult(this.result.getName());
+        this.testStructure.setResult(getResult().getName());
 
-        managers.testClassResult(this.result, null);
+        managers.testClassResult(getResult(), null);
```

Comment:
Really minor - do we need to call `getResult()` everywhere if we could instead use `this.resultData`?

Reply:
We went with a getter method here as Mike suggested that was a better way of coding it - I believe for unit testing purposes.